docs/source/model-registry.rst

```diff
@@ -78,47 +79,170 @@ version to one of the other valid stages.
 
 
 API Workflow
-------------
+============
+
+An alternative way to interact with Model Registry is using the `MLflow model flavor <https://www.mlflow.org/docs/latest/python_api/index.html>`_ or `MLflow Client Tracking API <https://www.mlflow.org/docs/latest/python_api/mlflow.tracking.html>`_ interface.
+In particular, you can register a model during an MLflow experiment run or after all your experiment runs.
+
+Adding an MLflow Model to the Model Registry
+--------------------------------------------
 
-You can register a model using two APIs: ``register_model`` and ``log_model``.
+There are three programmatic ways to add a model to the registry. First, you can use the ``mlflow.model_flavor.log_model(...)`` method. For example, in your code:
 
-If the model has already been logged, you can add it to the registry with the ``register_model`` API, referencing the model by its URI.
+.. code-block:: py
+
+    with mlflow.start_run(run_name="YOUR_RUN_NAME") as run:
+        params={"n_estimators": 5, "random_state": 42}
+        sk_learn_rfr=RandomForestRegressor(params)
+        # Log params using the MLflow sklearn flavor APIs
+        mlflow.log_params(params)
+        log_param("param_1", randint(0, 100))
+        log_metric("metric_1", random())
+        log_metric("metric_2", random() + 1)
+       …
+       # Log the sklearn model and register as version 1
+       mlflow.sklearn.log_model(sk_model=sk_learn_rfr,
+                artifact_path="sklearn-model",
+                registered_model_name="sk-learn-random-forest-reg-model")
+
+This logs the model as well as registers it under the specified name as version 1.
+
+The second way is to explicitly register the `mlflow.register_model(...) <https://www.mlflow.org/docs/latest/python_api/mlflow.html#mlflow.register_model>`_,
+after all your experiment runs and when you have ascertained which run within an experiment is most suitable to add to the registry.
+For this scheme, you will need the ``run_id`` as part of the ``runs:URI`` argument.
 
 .. code-block:: py
 
-   mlflow.register_model("runs:/e148278107274..832/artifacts/model", "EmailMarketingCampaign")
```

Comment:
Maybe we can name the arguments like above so it's clear what they are referring to

Reply:
I think "EmailMarketingCampaign" seems more like an Experiment name you want under which will want to create a set of different types of models—scikit-learn flavor, Keras or PyTorch etc. 

I suppose we can all it WeatherPowerForecast as general if you prefer.